client/app/waiting-time/waiting-time.component.ts

```diff
@@ -0,0 +1,42 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { timer } from 'rxjs';
+
+@Component({
+  selector: 'app-waiting-time',
+  templateUrl: './waiting-time.component.html',
+  styleUrls: ['./waiting-time.component.css']
+})
+export class WaitingTimeComponent implements OnInit {
+
+    waitingTime: string;
+    excessiveWaitingTime: boolean = false;
+
+    @Input() createdTime: string;
+
+    ngOnInit(){
+        //timer set to call get waiting time every second
+        timer(0,1000).subscribe(() => this.getWaitingTime());
+    }
+
+    /**
+     * Calculates time difference between current time and created time
+     */
+    getWaitingTime(){
+        let createdTime = new Date(this.createdTime).getTime();
```

Comment:
`createdTime` is already in milliseconds ?

Reply:
It wasn't working when I tried just doing this.createdTime.getTime().. even with the Input as a Date type. I console logged the typeof item.createdAt in the kitchen view and it was coming back as a string, so I just wrapped that string in a Date here. (hope that makes sense)

Reply:
Ahh makes sense, for some reason I thought the API returned it as milliseconds, dunno where i got that from.